priorityq.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import random
+
+class Item(object):
+
+    def __init__(self, value, priority):
+        self.value = value
+        self.priority = priority
+
+    # Priority with lower values are higher priority than those
+    # with higher values. E.g. Item with priority 1 is higher
+    # t
+    def higher_pri_than(self, item):
+        return self.priority < item.priority
+
+
+class PriorityQ(object):
+
+    def __init__(self):
+        self.pqheap = BinHeap()
+
+    def insert(self, item):
+        self.pqheap.push(item)
+
+    def pop(self):
+        return self.pqheap.pop()
+
+    def peek(self):
+        return self.pqheap.peek()
+
```

Comment:
The file could have stopped here if you would have just used the Binary Heap you made a few days ago.. 
I would give that a shot.